processor/src/main/java/com/umairjavid/kombind/processor/AdapterProcessor.kt

```diff
@@ -0,0 +1,129 @@
+package com.umairjavid.kombind.processor
+
+import com.google.auto.service.AutoService
+import com.squareup.kotlinpoet.ClassName
+import com.squareup.kotlinpoet.FileSpec
+import com.squareup.kotlinpoet.FunSpec
+import com.squareup.kotlinpoet.KModifier
+import com.squareup.kotlinpoet.ParameterSpec
+import com.squareup.kotlinpoet.ParameterizedTypeName.Companion.parameterizedBy
+import com.squareup.kotlinpoet.PropertySpec
+import com.squareup.kotlinpoet.TypeName
+import com.squareup.kotlinpoet.TypeSpec
+import com.squareup.kotlinpoet.asTypeName
+import com.umairjavid.kombind.anontation.SimpleKombindAdapter
+import java.io.File
+import javax.annotation.processing.AbstractProcessor
+import javax.annotation.processing.Messager
+import javax.annotation.processing.ProcessingEnvironment
+import javax.annotation.processing.Processor
+import javax.annotation.processing.RoundEnvironment
+import javax.lang.model.SourceVersion
+import javax.lang.model.element.TypeElement
+import javax.lang.model.element.VariableElement
+import javax.tools.Diagnostic
+import kotlin.reflect.KClass
+
+@AutoService(Processor::class)
+class AdapterProcessor : AbstractProcessor() {
+    lateinit var messenger: Messager
+    val genaratedClassSet = mutableSetOf<String>()
+
+    override fun init(env: ProcessingEnvironment?) {
+        super.init(env)
+        messenger = env!!.messager
+    }
+    override fun getSupportedAnnotationTypes() = mutableSetOf(SimpleKombindAdapter::class.java.name)
+
+    override fun getSupportedSourceVersion() = SourceVersion.latestSupported()
+
+    override fun process(typeElemnts: MutableSet<out TypeElement>?, roundEnv: RoundEnvironment?): Boolean {
+        val env = roundEnv!!.getElementsAnnotatedWith(SimpleKombindAdapter::class.java)
+        env.forEach {
+            val layoutRes = it.getAnnotation(SimpleKombindAdapter::class.java).layoutRes
+            if (it is VariableElement)
+                generateAdapterClass(it, layoutRes)
+        }
+        return false
+    }
+
+    fun generateAdapterClass(element: VariableElement, layoutRes: Int) {
+        if (!element.asType().asTypeName().isMutableLiveArraylist()) {
+            messenger.printMessage(Diagnostic.Kind.ERROR, "Needs to be of type MutableArrayList")
+            return
+        }
+
+        val kombindPackage = "com.umairjavid.kombind.ui"
+        val appPackageName = processingEnv.elementUtils.getPackageOf(element).simpleName.toString()
+        val kombindAdapterName = "KombindAdapter"
+        val viewHolder = "KombindAdapter.ViewHolder"
+        val elementName = element.simpleName
+        val enclosingClassName = element.enclosingElement.simpleName
+        val className = "Kombind${elementName.toString().capitalize()}Adapter_${enclosingClassName}"
```

Comment:
i think in that case it might be better to name it `KombindAdapter_${enclosingClassName.capitalize()}_${elementName.capitalize()}`